akka-actor-typed/src/main/scala/akka/actor/typed/internal/routing/GroupRouterImpl.scala

```diff
@@ -42,31 +42,20 @@ private final class GroupRouterImpl[T](
   routingLogic: RoutingLogic[T]
 ) extends AbstractBehavior[T] {
 
-  private var routees: Array[ActorRef[T]] = Array.empty[ActorRef[T]]
-
   // casting trix to avoid having to wrap incoming messages - note that this will cause problems if intercepting
   // messages to a router
   ctx.system.receptionist ! Receptionist.Subscribe(serviceKey, ctx.self.unsafeUpcast[Any].narrow[Receptionist.Listing])
+  private var routeesEmpty = true
 
   def onMessage(msg: T): Behavior[T] = msg match {
     case serviceKey.Listing(update) ⇒
       // we don't need to watch, because receptionist already does that
-      val newRoutees = update.toArray
-      if (newRoutees.nonEmpty) {
-        // make sure we keep a somewhat similar order, but don't put all entries from
-        // the same node next to each other
-        newRoutees.sortBy(ref ⇒ (ref.path.toStringWithoutAddress, ref.path.address))
-        routingLogic.routeesUpdated(routees, newRoutees)
-      }
-      routees = newRoutees
+      routingLogic.routeesUpdated(update.toList)
```

Comment:
`toVector` might be better

Reply:
or should we keep it as a `Set` in the API?

Reply:
Yeah, let's keep it a set, because that is what it is.